### ISSUE TITLE
[4b6] PHP 8 Issue - CLI array_merge() does not accept unknown named parameters

### ISSUE THREAD

**Issue by PhilETaylor** [joomla/joomla-cms#31923]
### Steps to reproduce the issue

Install Joomla 4 beta 6 on PHP 8.0.x

run command 

```
php cli/joomla.php -vvv
```

### Expected result

The standard Joomla command list of commands (This screenshot is with PHP 7.4 cli)

<img width="1090" alt="Screenshot 2021-01-08 at 22 11 27" src="https://user-images.githubusercontent.com/400092/104070315-803daa00-51fe-11eb-8d11-5774c3c30beb.png">

### Actual result

Error: `array_merge() does not accept unknown named parameters`

```
Joomla! 4.0.0-beta6 (debug: No)

Usage:
  command [options] [arguments]

Options:
  -h, --help            Display the help information
  -q, --quiet           Flag indicating that all output should be silenced
  -V, --version         Displays the application version
      --ansi            Force ANSI output
      --no-ansi         Disable ANSI output
  -n, --no-interaction  Flag to disable interacting with the user
  -v|vv|vvv, --verbose  Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug


In TextDescriptor.php line 172:

  [ArgumentCountError]
  array_merge() does not accept unknown named parameters


Exception trace:
  at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Descriptor/TextDescriptor.php:172
 array_merge() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Descriptor/TextDescriptor.php:172
 Joomla\Console\Descriptor\TextDescriptor->describeJoomlaApplication() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Descriptor/TextDescriptor.php:43
 Joomla\Console\Descriptor\TextDescriptor->describe() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Helper/DescriptorHelper.php:35
 Joomla\Console\Helper\DescriptorHelper->describe() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Command/ListCommand.php:73
 Joomla\Console\Command\ListCommand->doExecute() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Command/AbstractCommand.php:235
 Joomla\Console\Command\AbstractCommand->execute() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Application.php:1181
 Joomla\Console\Application->runCommand() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Application.php:403
 Joomla\Console\Application->doExecute() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/src/Application/ConsoleApplication.php:182
 Joomla\CMS\Application\ConsoleApplication->doExecute() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/vendor/joomla/console/src/Application.php:448
 Joomla\Console\Application->execute() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/libraries/src/Application/ConsoleApplication.php:225
 Joomla\CMS\Application\ConsoleApplication->execute() at /Users/phil/Downloads/Joomla_4.0.0-beta6-Beta-Full_Package/cli/joomla.php:69

list [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [--] [<command> [<namespace>]]
```

<img width="1483" alt="Screenshot 2021-01-08 at 22 10 49" src="https://user-images.githubusercontent.com/400092/104070253-61d7ae80-51fe-11eb-86b1-d55b4f967279.png">

**Comment by PhilETaylor** [comment]
A hint for you:
https://github.com/composer/composer/pull/9076/files

**Comment by alikon** [comment]
fixed here https://github.com/joomla-framework/console/pull/7

**Comment by wilsonge** [comment]
Fixed to 4.0-dev now with https://github.com/joomla/joomla-cms/commit/38c5ee51a96058d3264bde506ef077bb30324144
